### DISCUSSION TITLE
Next js without node js/add Next Js to an html page

### DISCUSSION THREAD

**Discussion by Pfurr** [vercel/next.js#25324] [Help]
Hi,  i'm working on a NOT React project. i'd like to add Next Js to an html page or other library (not node js). What is the best practice to do it? Could you give me some example of code? 

Thank you for your help

**Comment by scottschindler** [comment]
Next.js is a react framework used to create front-ends. You do not need a node backend to use Next.js, but you do need to have Node installed on your machine. 

It wouldn't make much sense to use Next.js for just one page of your front-end, rather you should use Next.js for your whole front-end.

Hope this helps!

**Reply by Pfurr** [reply]
hi @scottschindler, thank you for your help! I must to create a little components for an existing web site (cms in php, or similar). this component will have to add functionality in the front end and it will have to work ssr or not just client side

**Reply by scottschindler** [reply]
Ah, ok. Check out this blog post by someone from the Vercel team: https://nextjs.org/blog/incremental-adoption

**Reply by Pfurr** [reply]
@scottschindler thank you ;) I knew about this solution, but the problem is this: the component (or components) must be integrated into the customer's legacy web site (or store) and I just to provide the finished component that has to connect to the existing page (html css php ect...) on the same domain (example.com), into a singol page/portion of their website (they want to keep).
